### PR TITLE
WIP add grpc support

### PR DESCRIPTION
**Pull request by kozmagabor**
Add first version of grpc support with ssl and client cert config

### CODE REVIEW

File: dex/templates/deployment.yaml [hunk 1]
```diff
@@ -57,16 +57,28 @@ spec:
         resources:
           {{- toYaml .Values.resources | nindent 10 }}
         ports:
-          {{- toYaml .Values.ports | nindent 10 }}
+          - name: grpc
```
Comment (bonifaido) [inline dex/templates/deployment.yaml]: Please expose the telemetry port as well.

File: dex/templates/service.yaml [hunk 2]
```diff
@@ -15,11 +15,12 @@ spec:
   type: {{ .Values.service.type}}
   sessionAffinity: None
   ports:
-{{- range .Values.ports }}
-  - name: {{ .name }} 
-    port: {{ .containerPort }}
-    targetPort: {{ .containerPort}}
-{{- end}}
+  - name: http-grpc
```
Comment (bonifaido) [inline dex/templates/service.yaml]: Please expose the telemetry port as well.

File: dex/values.yaml [hunk 3]
```diff
@@ -76,13 +78,21 @@ config:
...
+    tls:
+      enabled: false
 
   grpc:
-    addr: ""
+    bind: 0.0.0.0
+    port: 5557
+    tls:
+      enabled: true
+      clientAuth: false
```
Comment (sagikazarmark) [inline dex/values.yaml]: `clientAuthEnabled`, `enableClientAuth` or `authenticateClient` would sound better IMHO.

Comment (kozmagabor) [reply dex/values.yaml]: you're right!

File: dex/templates/service.yaml (hunk 2 above)
Comment (kozmagabor) [reply dex/templates/service.yaml]: thx!

**Pull request closed without merging**